panel/widgets/remote_file_selector.py

```diff
@@ -0,0 +1,379 @@
+import asyncio
+import threading
+
+from abc import abstractmethod
+from pathlib import PurePosixPath
+from typing import (
+    ClassVar, Dict, List, Optional, Type,
+)
+
+import param
+
+from panel.layout import (
+    Column, Divider, ListPanel, Row,
+)
+
+from ..pane.markup import Markdown
+from ..viewable import Layoutable
+from .base import CompositeWidget
+from .button import Button
+from .input import TextInput
+from .select import CrossSelector
+
+
+class RemoteFileProvider(param.Parameterized):
+    """
+    `RemoteFileProvider` is an abstract class used by `RemoteFileSelector` to
+    list content of a remote filesystem.
+
+    Reference: https://panel.holoviz.org/reference/widgets/RemoteFileSelector.html
+
+    """
+    def __init__(self):
+        super().__init__()
```

Comment:
```suggestion
    def __init__(self, **params):
        super().__init__(**params)
```

Otherwise, parameters from inherited classes will not work. 

(The `__init__` is not needed here, but since this PR is still in draft, things can change)